Core/GDCore/Project/ObjectConfiguration.h

```diff
@@ -0,0 +1,189 @@
+   *
+   * \return false if the new value cannot be set
+   */
+  virtual bool UpdateProperty(const gd::String& name, const gd::String& value) {
+    return false;
+  };
+  ///@}
+
+  /** \name Drawing and editing initial instances
+   * Members functions related to drawing and editing initial instances of this
+   * object
+   */
+  ///@{
+  /**
+   * \brief Called when the IDE wants to know about the custom properties of an
+   * initial instance of this object.
+   *
+   * \return a std::map with properties names as key and values.
+   * \see gd::InitialInstance
+   */
+  virtual std::map<gd::String, gd::PropertyDescriptor>
+  GetInitialInstanceProperties(const gd::InitialInstance& instance,
+                               gd::Project& project,
+                               gd::Layout& layout);
+
+  /**
+   * \brief Called when the IDE wants to update a custom property of an initial
+   * instance of this object.
+   *
+   * \return false if the new value cannot be set
+   * \see gd::InitialInstance
+   */
+  virtual bool UpdateInitialInstanceProperty(gd::InitialInstance& instance,
+                                             const gd::String& name,
+                                             const gd::String& value,
+                                             gd::Project& project,
+                                             gd::Layout& layout) {
+    return false;
+  };
+    ///@}
+
+  /** \name Resources management
+   * Members functions related to managing resources used by the object
+   */
+  ///@{
+
+  /**
+   * \brief Called ( e.g. during compilation ) so as to inventory internal
+   * resources and sometimes update their filename. Implementation example:
+   * \code
+   * worker.ExposeImage(myImage);
+   * worker.ExposeFile(myResourceFile);
+   * \endcode
+   *
+   * \see ArbitraryResourceWorker
+   */
+  virtual void ExposeResources(gd::ArbitraryResourceWorker& worker) { return; };
+  ///@}
+
+/** \name Serialization
+ * Members functions related to serialization of the object
+ */
+///@{
+  /**
+   * \brief Serialize the object.
+   * \see DoSerializeTo
+   */
+  void SerializeTo(SerializerElement& element) const;
+
+  /**
+   * \brief Unserialize the object.
+   * \see DoUnserializeFrom
+   */
+  void UnserializeFrom(gd::Project& project, const SerializerElement& element);
+  ///@}
+
+ protected:
+  gd::String type;  ///< Which type is the object. ( To test if we can do
+                    ///< something reserved to some objects with it )
+
+  /**
+   * \brief Derived objects can redefine this method to load custom attributes.
+   */
+  virtual void DoUnserializeFrom(gd::Project& project,
+                                 const SerializerElement& element){};
+
+  /**
+   * \brief Derived objects can redefine this method to save custom attributes.
+   */
+  virtual void DoSerializeTo(SerializerElement& element) const {};
+};
+
+}  // namespace gd
+
+/**
+ * Objects are usually managed thanks to (smart) pointers.
```

Comment:
```suggestion
 * Object configurations are usually managed thanks to (smart) pointers.
```

But if this is used nowhere in the codebase, delete this.